userspace/engine/rule_loader_compiler.cpp

```diff
@@ -495,12 +495,11 @@ void rule_loader::compiler::compile_rule_infos(
 		}
 
 		// populate set of event types and emit an special warning
-		libsinsp::events::set<ppm_event_code> evttypes = { ppm_event_code::PPME_PLUGINEVENT_E };
 		if(rule.source == falco_common::syscall_source)
 		{
-			evttypes = libsinsp::filter::ast::ppm_event_codes(ast.get());
-			if ((evttypes.empty() || evttypes.size() > 100)
-			    && r.warn_evttypes)
+			auto evttypes = libsinsp::filter::ast::ppm_event_codes(ast.get());
+			evttypes.insert(ppm_event_code::PPME_ASYNCEVENT_E);
+			if ((evttypes.empty() || evttypes.size() > 100) && r.warn_evttypes)
```

Comment:
evvtypes.empty() can't ever be true here since we've just added PPME_ASYNCEVENT_E, right?

Reply:
Good catch, thanks!